### ISSUE TITLE
About the file matlab.tmLanguage

### ISSUE THREAD

**Issue by zhaouv** [Gimly/vscode-matlab#77]
I don't know whether using a file belonging to another repository is a good idea or not.#74  
The repository was updated one year ago.  
And unfortunately the highlight is still wrong.  
#58 #69 
+ now
![image](https://user-images.githubusercontent.com/26158289/55265595-61e80380-524f-11e9-8395-52091e93b9af.png)
+ default IDE
![image](https://user-images.githubusercontent.com/26158289/55265618-79bf8780-524f-11e9-89c3-5fca6b6b99ab.png)
+ fixed  
I fix it and then I find your start using a file out of the repository..  
Maybe I should send a pr to that repository.  
![image](https://user-images.githubusercontent.com/26158289/55265842-6365fb80-5250-11e9-9dad-74b77fb0bde7.png)

**Comment by zhaouv** [comment]
https://github.com/mathworks/MATLAB-Language-grammar/pull/1  
I don’t how long it will take for their merging

**Comment by zhaouv** [comment]
I did not test other file for the new `matlab.tmLanguage`.   
It seems there are more bugs in other cases.#76  
You can use the old file, I will fix the known bug for it.
The File in `mathworks/MATLAB-Language-grammar` seems too old.

**Comment by Gimly** [comment]
Yeah, I agree with you, I didn't test it enough before accepting the PR. Since it was coming from Mathworks I thought it would fix all the bugs that I had with the syntax grammar, but I see that even though it did fix a few issues, it didn't fix everything.

Unfortunately these days I don't work a lot with Matlab anymore, so I don't use this extension a lot, I rely on the community to send me the bugs.

**Comment by zhaouv** [comment]
https://gist.github.com/zhaouv/077476e20386843fade406806b9dac26  

I put to fixed file in the gist.
It bases on the [branch](https://github.com/Gimly/vscode-matlab/tree/820cd9f51b235932470ec38a46f12bb59c9c4d0e).  
And these issues are fixed [#69](https://github.com/Gimly/vscode-matlab/issues/69), [#31](https://github.com/Gimly/vscode-matlab/issues/31).  

About this issue [#53](https://github.com/Gimly/vscode-matlab/issues/53).  
It seems Matlab supports `"` to quote a string in the new version. I don't know the behavior of `"-string`. So I can not write the TextMate rule of it. In R2016b, there is no bug here.

**Comment by zhaouv** [comment]
#53 fixed (by guessing) https://gist.github.com/zhaouv/077476e20386843fade406806b9dac26/revisions  
Suppose the two quotes behave consistently...  
![image](https://user-images.githubusercontent.com/26158289/55747630-ae2f0280-5a0a-11e9-9e83-2c7e257a1c9f.png)

**Comment by zhaouv** [comment]
@gauravsavadi  
Which version should we start from?  
the old file in vscode-matlab is about 94kb
the file in MATLAB-Language-grammar is about 32kb

**Comment by Gimly** [comment]
@zhaouv which issues is there left on this topic? Looks like both line comments and nested comments issues are fixed. Quotes as well. Is there something in here open?

**Comment by zhaouv** [comment]
I didn't test the file in gauravsavadi/MATLAB-Language-grammar. (It seems they do not want to highlight functions.)  
For the old file, all known highlighting bugs are fixed.
I can also test and fix bugs in gauravsavadi/MATLAB-Language-grammar, if you want to use this file.

**Comment by gauravsavadi** [comment]
@rlivings39 Is VSCode reusing the MathWorks grammar? The repository at https://github.com/mathworks/MATLAB-Language-grammar should be up to date with all the fixes after we've merged the PRs?

**Comment by Gimly** [comment]
@gauravsavadi we're now using the MathWorks grammar, it's configured as a submodule and I've updated it yesterday.

@zhaouv the functions highlighting issue has been fixed, we now have another grammar configuration file in https://github.com/Gimly/vscode-matlab/tree/master/syntaxes which contains only the functions highlighting (and eventually in the future other things that would not be merged to MathWork's grammar).

I did a few tests and it seems like most, if not all highlighting issues we had have already been fixed in MathWorks grammar, but feel free to test it further. Thanks a lot for your work and help.
